lib/hexpm/utils.ex

```diff
@@ -281,4 +281,8 @@ defmodule Hexpm.Utils do
   def datetime_to_rfc2822(%DateTime{calendar: Calendar.ISO, time_zone: "Etc/UTC"} = datetime) do
     Calendar.strftime(datetime, "%a, %d %b %Y %H:%M:%S GMT")
   end
+
+  def latest_valid_release_with_docs(releases) do
+    Enum.find(releases, &(&1.has_docs && &1.version.pre == []))
```

Comment:
It's unfortunately not this simple, for example what if a package only has pre-releases. You can extend this function with a `:with_docs` option: https://github.com/hexpm/hexpm/blob/main/lib/hexpm/repository/release.ex#L189.